3-enrich/scala-hadoop-bad-rows/src/main/scala/com/snowplowanalytics/hadoop/scalding/JsProcessor.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2016 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.hadoop.scalding
+
+import org.mozilla.javascript._
+
+import scala.util.control.NonFatal
+
+object JsProcessor {
+
+  object Variables {
+    private val prefix = "$snowplow31337" // To avoid collisions
+    val InTsv  = s"${prefix}InTsv"
+    val InErrors  = s"${prefix}InErrors"
+    val Out = s"${prefix}Out"
+  }
+
+  /**
+   * Create a script containing the user-defined "process" function,
+   * some helper functions, and a call to the "process" function
+   *
+   * @param sourceCode The JavaScript provided by the user
+   * @return Compiled script
+   */
+  def compile(sourceCode: String): Script = {
+    val wholeScript =
+      s"""|// Helper functions
+          |function tsvToArray(event) {
+          |  return event.split('\t', -1);
+          |}
+          |function arrayToTsv(tsv) {
+          |  return tsv.join('\t');
+          |}
+          |
```

Comment:
These helpers should be documented in the wiki documentation for Hadoop Event Recovery.